tonic-build/src/client.rs

```diff
@@ -22,19 +23,34 @@ pub fn generate<T: Service>(
 
     let struct_debug = format!("{} {{{{ ... }}}}", &service_ident);
 
+    let package = if emit_package { service.package() } else { "" };
+    let path = format!(
+        "{}{}{}",
+        package,
+        if package.is_empty() { "" } else { "." },
+        service.identifier()
+    );
+
+    let mod_attributes = attributes.for_mod(package);
+    let struct_attributes = attributes.for_struct(&path);
+    let impl_attributes = attributes.for_impl(&path);
```

Comment:
is impl attributes actually useful?

Reply:
the more i think about it, probably not -- any use case would likely be better served by writing a `prost_build::ServiceGenerator`

Reply:
I think we should just keep mod and struct ones and then I think i'm ok to merge this!